lib/stdlib/src/string.erl

```diff
@@ -1146,6 +1146,14 @@ length_1(Str, N) ->
         {error, Err} -> error({badarg, Err})
     end.
 
+length_b(<<CP2, CP3, CP4, CP5, CP6, CP7, CP8, CP9, Rest/binary>>,
+         CP1, N)
+  when CP1 =/= $\r,CP2 =/= $\r,CP3 =/= $\r,CP4 =/= $\r,
+       CP5 =/= $\r,CP6 =/= $\r,CP7 =/= $\r, CP9 =/= $\r,
+       CP2 < 128, CP2 < 128, CP3 < 128, CP4 < 128,
+       CP5 < 128, CP6 < 128, CP7 < 128, CP8 < 128,
+       CP9 < 128 ->
+    length_b(Rest, CP8, N+8);
```

Comment:
This seems like an unrelated change to the base subject of the PR. Why is it necessary? Was `string:length` incorrect before?

Reply:
It was slow, and was used a lot by earlier commits, the code is now more optimized,
so maybe should be moved out of this PR.

Reply:
Ah, right. Now reading it again I noticed it's an optimisation. Perhaps just a separate commit in this PR with a small comment that this is an optimisation would work?